qemu/tests/timerdevice_boot.py

```diff
@@ -141,19 +147,18 @@ def update_clksrc(session, clksrc):
     if params["os_type"] == "linux":
         error_context.context("Check the hardware time on guest", logging.info)
         guest_hwtime = get_hwtime(session)
+        drift = abs(float(host_time) - float(guest_hwtime))
+        if drift > timerdevice_drift_threshold:
+            test.fail("The guest's hardware time is different with"
+                      " host's system time. Host time: '%s', guest time:"
+                      " '%s'" % (host_time, guest_hwtime))
 
     drift = abs(float(host_time) - float(guest_time))
     if drift > timerdevice_drift_threshold:
         test.fail("The guest's system time is different with"
                   " host's. Host time: '%s', guest time:"
                   " '%s'" % (host_time, guest_time))
```

Comment:
I think it's a good choice to wrap a function to check guest system time and hardware time. Thanks!